### PR TITLE
Go resilient services

### PR DESCRIPTION
**Pull request by katcipis**
First version of the presentation

### CODE REVIEW

File: golang/resilient-services-in-go/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Resilient Services in Go
+
+## Abstract
+
+Resilience is not about never failing, but how do you recover from it.
+How you can prevent your services from locking down or exhausting all
```
Comment (vitorarins) [inline golang/resilient-services-in-go/README.md:6]: I think it should be: `How can you...`

Comment (katcipis) [reply golang/resilient-services-in-go/README.md:6]: Thanks :-)

File: golang/resilient-services-in-go/resilientservices.slide [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+- What to do when your test is expensive ?
+- Go routine pools with timeouts
+- Error handling
+- Final thoughts
+
+
+* Recovery Oriented Computing
+
+
+It is not a mater of *if it will fail* but *what will happen when it fails*.
```
Comment (vitorarins) [inline golang/resilient-services-in-go/resilientservices.slide]: Typo: `matter`

**Comment by vitorarins** [comment]
Great ideas ;)

File: golang/resilient-services-in-go/resilientservices.slide (hunk 2 above)
Comment (katcipis) [reply golang/resilient-services-in-go/resilientservices.slide]: Great :-)

File: golang/resilient-services-in-go/resilientservices.slide [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+* context.Context
+
+- Moved to Go core in 1.7
+- Models timeouts and cancellation
+
+
+* Timeout Context
+
+.play examples/timeout.go
```
Comment (i4ki) [inline golang/resilient-services-in-go/resilientservices.slide]: The example calls _cancel()_  independently of the job being done or not. Sorry if it's exactly what you want to show, but I would expect an example that cancels only if the timeout exceeded. Like the godoc example below:

https://golang.org/pkg/context/#example_WithTimeout

File: golang/resilient-services-in-go/examples/expiration.go [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+func main() {
+	ctx, cancel := context.WithTimeout(
+		context.Background(),
+		10*time.Second,
+	)
+	defer cancel()
+
+	fmt.Println("Waiting for timeout")
+	<-ctx.Done()
+	fmt.Println("Done")
```
Comment (i4ki) [inline golang/resilient-services-in-go/examples/expiration.go]: Maybe print the ctx.Err() to make clear the timeout exceeded:

``` sh
λ> go run exp.go
Waiting for timeout
context deadline exceeded
```

File: golang/resilient-services-in-go/resilientservices.slide [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+* Timeout Context
+
+- Will be canceled when timeout expires
+- Or when cancel is called
+
+
+* Waiting to expire
+
+.play examples/expiration.go
```
Comment (i4ki) [inline golang/resilient-services-in-go/resilientservices.slide]: Sorry, the example is here... Ignore the last review.

**Comment by katcipis** [comment]
@kamilash @vitorarins @tiago4orion It is done now, review it if possible :-)

File: golang/resilient-services-in-go/examples/expiration.go (hunk 4 above)
Comment (katcipis) [reply golang/resilient-services-in-go/examples/expiration.go]: Added, thanks :-)

**Comment by kamilash** [comment]
Congratulations! I bet it will be a really good presentation! :D

**Pull request merged**